includes/class-support-relay.php

```diff
@@ -0,0 +1,147 @@
+<?php
+if ( !defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+	/**
+	 * Blog_Tutor_Support helper class
+	 *
+	 * @package  Blog_Tutor_Support
+	 * @category Core
+	 * @author Sergio Scabuzzo
+	 */
+
+class NerdPress_Support_Relay {
+
+	public static function init() {
+		$class = __CLASS__;
+		new $class;
+	}
+
+	public function __construct() {
+		add_action( 'wp_loaded', array( $this, 'ping_relay' ) );
+	}
+
+	/**
+	 * Send a request to the relay without an HTTP request
+	 *
+	 * @since 1.0.0
+	 */
+	public static function ping_relay_headless() {
+
+		$dump         = self::assemble_dump();
+		$api_response = self::send_request_to_relay( $dump );
+
+	}
+
+	/**
+	 * Ping the relay server if the PING is set in the GET request.
+	 *
+	 * @since 0.8.2
+	 */
+	public static function ping_relay() {
+
+		// If the request is a one-time call from the dashboard.
+		if ( isset( $_GET['ping'] ) ) {
+
+			$dump = self::assemble_dump();
+
+			$api_response = self::send_request_to_relay( $dump );
+
+			if ( $api_response['response']['code'] === 201 ) {
+				nocache_headers();
```

Comment:
Move these two lines outside the if statement and log the error. So we can redirect (remove the ?ping) no matter the API result.